### PR TITLE
chores: add Build Server debug config

### PR DESCRIPTION
**Pull request by Jiaaming**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,6 +40,32 @@ The extension uses a Gradle plugin (`com.microsoft.gradle.GradlePlugin`) to get
...
+     "hostName": "localhost",
+     "port": "8989",
+     "projectName": "server"
+   }
+   ```
+
+3. In your main project (vscode-gradle), start the `Debug Extension & Build Server` launch configuration.
+
+4. In the build-server-for-gradle project, start the `Attach to Gradle Build Server` launch configuration.
+
```
Comment (jdneo) [inline CONTRIBUTING.md:61]: Just leave one empty line is enough

**Comment by Jiaaming** [comment]
> @Jiaaming please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: .vscode/launch.json [hunk 2]
```diff
@@ -81,6 +81,26 @@
...
+      "outFiles": [
+        "${workspaceFolder}/extension/dist/**/*.js"
+      ],
+      "preLaunchTask": "Gradle: Build",
+      "presentation": {
+        "group": "debug",
+        "order": 3
+      },
+      "env": {
+        "DEBUG_VSCODE_BUILD_SERVER":"true"
```
Comment (jdneo) [inline .vscode/launch.json]: `DEBUG_VSCODE_BUILD_SERVER` -> `DEBUG_GRADLE_BUILD_SERVER`

(and for all other occurrence)

File: extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java [hunk 3]
```diff
@@ -84,16 +86,23 @@ public static BuildServerConnection getBuildServerConnection(IPath rootPath) thr
...
-                "-Dplugin.dir=" + pluginPath,
-                "-cp",
-                String.join(getClasspathSeparator(), classpaths),
-                "com.microsoft.java.bs.core.Launcher"
-        );
+        String debugServer = System.getenv("DEBUG_VSCODE_BUILD_SERVER");
+        List<String> command = new ArrayList<>();
+
+        command.add(javaExecutablePath);
+        if ("true".equals(debugServer)) {
```
Comment (jdneo) [inline extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java]: Or you can simplify it to:

```java
if (Boolean.parseBoolean(System.getenv("DEBUG_GRADLE_BUILD_SERVER"))) {
    command.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=8989");
}
```

**Review by jdneo — COMMENTED** [review]
LGTM, just some minor issues.

File: extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java [hunk 4]
```diff
@@ -84,16 +86,23 @@ public static BuildServerConnection getBuildServerConnection(IPath rootPath) thr
...
+            command.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=8989");
+        }
+        command.add("--add-opens=java.base/java.lang=ALL-UNNAMED");
+        command.add("--add-opens=java.base/java.io=ALL-UNNAMED");
+        command.add("--add-opens=java.base/java.util=ALL-UNNAMED");
+        command.add("-Dplugin.dir=" + pluginPath);
+        command.add("-cp");
+        command.add(String.join(getClasspathSeparator(), classpaths));
+        command.add("com.microsoft.java.bs.core.Launcher");
+
```
Comment (jdneo) [inline extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java:101]: nit: Just leave one empty line

File: extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java [hunk 5]
```diff
@@ -84,16 +86,23 @@ public static BuildServerConnection getBuildServerConnection(IPath rootPath) thr
...
-                "--add-opens=java.base/java.io=ALL-UNNAMED",
-                "--add-opens=java.base/java.util=ALL-UNNAMED",
-                "-Dplugin.dir=" + pluginPath,
-                "-cp",
-                String.join(getClasspathSeparator(), classpaths),
-                "com.microsoft.java.bs.core.Launcher"
-        );
+
+        List<String> command = new ArrayList<>();
+
```
Comment (jdneo) [inline extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/ImporterPlugin.java]: nit: This empty line is no needed.

**Review by jdneo — APPROVED** [review]
LGTM! Thank you @Jiaaming for your contribution!

**Pull request merged**